### PR TITLE
Only one save button in threat modal

### PR DESCRIPTION
**Pull request by ghost91-**
* Removes the redundant "Save" button from the threat modal (fields are saved anyways when they loose focus).
* Renames the "Save & Add" / "Save & Update" button to just "Save" and puts it into the location of where the previous "Save" button has been
* Removes all buttons from the threat modal for non-owner players.

### CODE REVIEW

File: src/client/components/threatmodal/threatmodal.js [hunk 1]
```diff
@@ -103,35 +98,34 @@ class ThreatModal extends React.Component {
...
-          <ModalFooter>
-            <Button color="success" className="mr-auto" hidden={this.props.G.threat.owner !== this.props.playerID} onClick={() => this.saveThreat()}>Save</Button>{' '}
-            <Button color="primary" disabled={this.isInvalid()} onClick={() => this.addOrUpdate()}>{(this.props.G.threat.new ? 'Save & Add' : 'Save & Update')}</Button>{' '}
-            <Button color="secondary" disabled={this.props.G.threat.owner !== this.props.playerID} onClick={() => this.props.moves.toggleModal()}>Cancel</Button>
-          </ModalFooter>
-          <ModalFooter hidden={this.props.G.threat.owner !== this.props.playerID}>
-            <small className="mr-auto text-muted"><b>TIP:</b> Saving would allow other players to view your changes instantly.</small>
-          </ModalFooter>
+          {
+            this.isOwner &&
```
Comment (ChristophNiehoff) [inline src/client/components/threatmodal/threatmodal.js:123]: A general question: Is it better to use the `hidden` parameter or conditional rendering for these reactstrap components? At the moment, both is used in this file...
We should make it consistent.

**Review by ChristophNiehoff — CHANGES_REQUESTED** [review]
This feature is great and looks pretty cool!
Just a few coding style remarks.

File: src/client/components/threatmodal/threatmodal.js [hunk 2]
```diff
@@ -103,35 +98,34 @@ class ThreatModal extends React.Component {
...
                 <option>Medium</option>
                 <option>High</option>
               </Input>
             </FormGroup>
             <FormGroup>
               <Label for="description">Description</Label>
-              <Input type="textarea" name="description" id="description" disabled={this.props.G.threat.owner !== this.props.playerID} style={{ height: 150 }} value={this.state.description} onBlur={(e) => this.props.moves.updateThreat("description", e.target.value)} onChange={(e) => this.updateState("description", e.target.value)} />
+              <Input type="textarea" name="description" id="description" disabled={!this.isOwner} style={{ height: 150 }} value={this.state.description} onBlur={(e) => this.props.moves.updateThreat("description", e.target.value)} onChange={(e) => this.setState({description: e.target.value})} />
             </FormGroup>
             <FormGroup hidden={this.props.G.threat.owner !== this.props.playerID}>
```
[the remark this answers is not in the document]
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js:111]: yeah, I missed that one.

File: src/client/components/threatmodal/threatmodal.js [hunk 3]
```diff
@@ -103,35 +98,34 @@ class ThreatModal extends React.Component {
...
               <div className="checkbox-item">
-                <Input className="pointer" type="checkbox" id="showMitigation" onChange={(e) => this.toggleMitigationForm(e.target.checked)} />
+                <Input className="pointer" type="checkbox" id="showMitigation" onChange={(e) => this.toggleMitigationField(e.target.checked)} />
                 <Label for="showMitigation">Add a mitigation <em>(optional)</em></Label>
               </div>
             </FormGroup>
-            <FormGroup hidden={this.props.G.threat.owner === this.props.playerID && !this.state.showMigitation}>
+            <FormGroup hidden={this.props.G.threat.owner === this.props.playerID && !this.state.showMitigation}>
```
[the remark this answers is not in the document]
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js:117]: it shouldn't be negated but yeah, I seem to have missed this one.

File: src/client/components/threatmodal/threatmodal.js [hunk 4]
```diff
@@ -54,32 +53,28 @@ class ThreatModal extends React.Component {
     // update the values from the state
     this.saveThreat();
```
[the remark this answers is not in the document]
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js:54]: I didn't want to dig too deep into this yet. Still figuring out what all these things do. But I agree that the naming could be much clearer. I also think that we might not actually need all of the functions if things are properly refactored. But I would prefer to leave this as-is for now and adjust it when I properly understand it.

File: src/client/components/threatmodal/threatmodal.js (hunk 1 above)
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js:123]: I think I would prefer to use conditional rendering. If they are just hidden, they will still be in the DOM, which doesn't make a lot of sense, IMO.

**Comment by ChristophNiehoff** [comment]
And another general comment: You did not add any tests  ;)  Could you please add some tests that the correct elements vanish for the different players? And maybe also that the buttons trigger the right actions (or do not trigger for the cancel button)?

**Comment by ghost91-** [comment]
> And another general comment: You did not add any tests ;) Could you please add some tests that the correct elements vanish for the different players? And maybe also that the buttons trigger the right actions (or do not trigger for the cancel button)?

good idea!

File: src/client/components/threatmodal/threatmodal.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js]: addressed

File: src/client/components/threatmodal/threatmodal.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ghost91-) [reply src/client/components/threatmodal/threatmodal.js]: addressed

**Review by ChristophNiehoff — APPROVED** [review]
Looks great! Thanks for the large number of tests!

**Pull request merged**